### PR TITLE
Warn about non-explicit transaction fees

### PR DESCRIPTION
**Pull request by morleyzhi**
If a user calls `TransactionBuilder` without a fee, issue a console log. Also update code examples to provide a fee, and export BASE_FEE. Part of #166.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ A breaking change will get clearly notified in this log.
 
 ## In `master`
 
+- _Warning_ Calling TransactionBuilder without a `fee` param is now deprecated
+  and will issue a warning. In a later release, it throw an error.
```
Comment (ire-and-curses) [inline CHANGELOG.md]: In a later release, it **will** throw an error.

File: src/transaction_builder.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { Keypair } from './keypair';
 import { Transaction } from './transaction';
 import { Memo } from './memo';
 
-const BASE_FEE = 100; // Stroops
+export const BASE_FEE = 100; // Stroops
```
Comment (ire-and-curses) [inline src/transaction_builder.js]: Any merit in relaxing the constant nature of this until we implement [#233](https://github.com/stellar/js-stellar-sdk/issues/233)? I saw a comment earlier about users having to hack the hard-coded value in the SDK.

File: src/transaction_builder.js [hunk 3]
```diff
@@ -73,6 +73,13 @@ export class TransactionBuilder {
     }
     this.source = sourceAccount;
     this.operations = [];
+
+    if (isUndefined(opts.fee)) {
+      console.log(
+        '[TransactionBuilder] The `fee` parameter of `TransactionBuilder` is required. Future versions of this library will error if not provided.'
```
Comment (bartekn) [inline src/transaction_builder.js]: Maybe add a package name here? Will be helpful to find what generated this message in larger projects.

File: src/transaction_builder.js (hunk 2 above)
Comment (morleyzhi) [reply src/transaction_builder.js]: Users shouldn't change this value; if they don't want to use it, they should use their own constant or variable.

File: src/transaction_builder.js (hunk 3 above)
Comment (morleyzhi) [reply src/transaction_builder.js]: The challenge I was trying to work around is that most people would import `stellar-sdk`, and a few would import `stellar-base`, and we don't know at this juncture which they're importing. So I think it's better to be vague and let the dev make the connection, vs. naming the wrong one or naming both.

File: CHANGELOG.md (hunk 1 above)
Comment (morleyzhi) [reply CHANGELOG.md]: Fixed!

**Pull request merged**